tutugodfrey/Pro-tutor
# #161571776 As a user I should be able to sign up

#### What does this pull request do?
- Add implementation for user signup with in-memory data
#### Description of tasks to be completed?
- Define the signup route.
- Create a users controller with the signup method.
- Write tests to ensure functionality
- Ensure all required fields are validated

#### How should this be manually tested?
- Make a copy of the branch, install dependencies and run npm `run start:dev` to start the development server
- Try creating a user using the route `http://localhost:500/apiv1/signup with the schema below. Fill in the fields. Also, try creating a user with invalid data and confirm that you get results similar to the screenshot below.

```User schema```
{
	"firstName": "",
	"lastName": "",
	"email": "",
	"password": "",
	"username":"",
	"confirmPassword": ""
}
#### Screenshots
![image](https://user-images.githubusercontent.com/30448913/52527743-06c57600-2cce-11e9-9563-ca97eb5925c6.png)

![image](https://user-images.githubusercontent.com/30448913/52527747-10e77480-2cce-11e9-81af-76d635e55187.png)

#### What are the Pivotal Tracker stories
#161571776
